apps/service_define/test/define/typespec_analysis_test.exs

```diff
@@ -0,0 +1,23 @@
+defmodule Define.TypespecAnalysisTest do
+  use ExUnit.Case
+  alias Define.{TypespecAnalysis}
+
+  test "get_types() returns all types from the typespec" do
+    expected = %{
+      "integer_type" => "integer",
+      "string_type" => "string",
+      "atom_type" => "atom",
+      "float_type" => "float",
+      "string_list_type" => {"list", "string"},
+      "integer_list_type" => {"list", "integer"},
+      "map_type" => "map",
+      "dictionary_type" => "dictionary"
+    }
+
+    assert expected == TypespecAnalysis.get_types(TypespecAnalysisTestModule)
+  end
+
+  test "returns empty map when a typespec is not defined" do
+    assert %{} == TypespecAnalysis.get_types(TypespecAnalysisTestModuleNoSpec)
+  end
```

Comment:
```suggestion
  describe "get_types/1" do
    test "returns all types from the typespec" do
      expected = %{
        "integer_type" => "integer",
        "string_type" => "string",
        "atom_type" => "atom",
        "float_type" => "float",
        "string_list_type" => {"list", "string"},
        "integer_list_type" => {"list", "integer"},
        "map_type" => "map",
        "dictionary_type" => "dictionary"
      }

      assert expected == TypespecAnalysis.get_types(TypespecAnalysisTestModule)
    end

    test "returns empty map when a typespec is not defined" do
      assert %{} == TypespecAnalysis.get_types(TypespecAnalysisTestModuleNoSpec)
    end
  end
```

This is more or less the pattern we're following for grouping tests around a single function.